Reaktoro/Core/SpeciesList.hpp

```diff
@@ -196,25 +196,55 @@ class SpeciesListBase
     /// Return all species with a given tag.
     auto withTag(String tag) const -> SpeciesList
     {
-        return filter(m_species, RKT_LAMBDA(s, contains(s.tags(), tag)));
+        // Filter only if the tag is not empty, otherwise return the same list of species
+        if(tag.size())
```

Comment:
I'm correcting `contains` so that we do not have to test here again if `s.tags()` is empty. This is now tested in `contains`.